test/object-store/frozen_objects.cpp

```diff
@@ -289,6 +338,50 @@ TEST_CASE("Freeze Results", "[freeze_results]") {
         realm->close();
         REQUIRE(DB::call_with_lock(config.path, [](auto) {}));
     }
+
+    SECTION("Results after source remove") {
+        Results results;
+        std::function<void()> do_remove;
+
+        SECTION("Results on collection") {
+            Obj obj;
+            write([&]() {
+                obj = create_object(42, true);
+            });
+
+            SECTION("Dictionary") {
+                results = Results(realm, obj.get_dictionary_ptr(int_dict_col));
+            }
+
+            SECTION("Links") {
```

Comment:
This addition was actually needed to reproduce the crash as in bug report - in essence to test that collection->is_attached should be checked first